src/commands.ts

```diff
@@ -0,0 +1,168 @@
+import * as strings from './strings';
+import { DotNetInfo, requireDotNetSdk, runDotNetCommand, findProjectTemplate } from './dotnet';
+import { CommandObserver } from './CommandObserver';
+import * as Constants from './constants';
+
+export default function registerCommands(commandObserver: CommandObserver, packageInfo: Utils.IPackageInfo): vscode.Disposable[] {
+    let dotNetSdkVersion = packageInfo.requiredDotNetCoreSDK;
+    return [
+        vscode.commands.registerCommand('pgproj.build.all', async () => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await vscode.window.withProgress({
+                        location: vscode.ProgressLocation.Notification,
+                        title: Constants.buildProgressTitle,
+                        cancellable: true
+                    }, async (progress, token) => {
+                        await buildAllProjects(dotnet, commandObserver, token);
+                });
+            });
+        }),
+        vscode.commands.registerCommand('pgproj.build.current', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await vscode.window.withProgress({
+                        location: vscode.ProgressLocation.Notification,
+                        title: Constants.buildProgressTitle,
+                        cancellable: true
+                    }, async (progress, token) => {
+                        await buildCurrentProject(args, dotnet, commandObserver, token);
+                    });
+                });
+        }),
+        vscode.commands.registerCommand('pgproj.add.new', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await addNewPostgreSQLProject(args, dotnet, packageInfo.projectTemplateNugetId, commandObserver);
+            })
+        }),
+    ];
+}
+
+async function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
+    let project = '';
+    if (!args) {
+        const activeEditor = vscode.window.activeTextEditor;
+        if (activeEditor !== undefined) {
+            project = activeEditor.document.uri.fsPath;
+        }
+    } else {
+        project = args.fsPath;
+    }
+
+    commandObserver.clear();
+
+    let isValid = await validateProjectSDK([project], commandObserver);
+    if (!isValid) {
+        return;
+    }
+    await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    try {
+        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+        commandObserver.clear();
+
+        let isValid = await validateProjectSDK(projects.map(p => p.fsPath), commandObserver);
```

Comment:
I'm thinking if it will be better user experience if we just build all projects on supported SDK and skip (producing an error message) for those that are not supported. The code will be streamlined too - you can just have one method to check and build and use it in both commands, rather than duplicating the validation call.

Reply:
So you mean like just skip those projects all together? and just print an output that these projects were skipped?